### PR TITLE
feat: README for the Valve Monitor accelerator

### PR DESCRIPTION
**Pull request by tjvantoll**
This PR updates all of the various READMEs for the valve monitor accelerator.

### CODE REVIEW

**Comment by tjvantoll** [comment]
@ethanpierce Can you take a look over this section and tell me if anything seems off? https://github.com/blues/app-accelerators/tree/tj-nf9-readme/valve-monitor#hardware

**Comment by tjvantoll** [comment]
@haydenroche5 In this PR I’m updating the NF9 firmware docs to be consistent with the other accelerators in this repo. See https://github.com/blues/app-accelerators/blob/tj-nf9-readme/valve-monitor/firmware/README.md. Let me know if anything looks off or if you have any concerns.

File: valve-monitor/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+Additionally you may wish to also use the following.
+
+* (Optional) A manual valve such as this [YZM Quick Connector](https://www.amazon.com/dp/B077H2JWSZ/ref=cm_sw_r_api_i_38J4S2VWFKAZBVEA8GCM_0)
+to make it easier to start/stop the flow of liquid while testing.
+* (Optional) An enclosure for your hardware such as this
+[outdoor-friendly enclosure from Sixfab](https://sixfab.com/product/raspberry-pi-ip54-outdoor-iot-project-enclosure/).
+
+Once you have all of your hardware you’ll next need to assemble the pieces.
+As a start, complete the [Notecard and Notecarrier-F quickstart guide](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/), which will help
```
Comment (paigen11) [inline valve-monitor/README.md:91]:
```suggestion
To start, complete the [Notecard and Notecarrier-F quickstart guide](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-f/), which will help
```

File: valve-monitor/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+for supplying correct voltage to this project’s various components.
+* An I2C level shifter, such as [this one from HiLetgo](https://www.amazon.com/dp/B07F7W91LC/ref=cm_sw_r_api_i_FZPJ7VRY2329ARNQ3W22_0), to
+translate the flow meter’s pulses from 5V logic to 3.3V logic for the Swan
+microcontroller.
+* A MOSFET driver, such as this one from [HiLetgo](https://www.amazon.com/dp/B01I1J14MO/ref=cm_sw_r_api_i_8YRY25Q7R9HGV1ZPHERP_0),
+is required to allow the Swan to switch on high-power devices such as the
+solenoid, without needing to send that power from the microcontroller. (The
+Swan can only send about 0.066 Watts (3.3V*20mA) out of a GPIO.)
+
+Additionally you may wish to also use the following.
```
Comment (paigen11) [inline valve-monitor/README.md:83]:
```suggestion
Additionally you may wish to also use the following:
```

File: valve-monitor/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+![The Notecard, Notecarrier, and Swan connected](images/notecard-and-notecarrier.jpg)
+
+From there you’ll need to connect both your flow meter and solenoid valve
+to the Notecarrier. If you’re using this project’s recommended hardware, here’s a [wiring diagram that shows how all the components connect](https://miro.com/app/board/uXjVPL8v4hE=/).
+
+![Wiring diagram of the various valve monitor components](images/wiring-diagram.png)
+
+Regardless of which hardware you use, you’ll need to have the following pins
+wired to your Notecarrier.
```
Comment (paigen11) [inline valve-monitor/README.md:102]:
```suggestion
wired to your Notecarrier:
```

File: valve-monitor/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+pin. The Swan uses this pin to read the flow rate from the flow meter.
+* The Notecarrier’s `F_D13` pin must be wired to the Notecarrier’s `ATTN` pin. The
+Swan uses this to [respond to interrupts](https://dev.blues.io/guides-and-tutorials/notecard-guides/attention-pin-guide/)
+whenever the Notecard receives a command to open or close the solenoid valve.
+
+When everything is put together your build should look something like this.
+
+![The Notecarrier connected to the solenoid and the valve monitor](images/full-build-no-case.jpg)
+
+If you’re additionally using a case like the [recommended one from Sixfab](https://sixfab.com/product/raspberry-pi-ip54-outdoor-iot-project-enclosure/),
```
Comment (paigen11) [inline valve-monitor/README.md:116]:
```suggestion
If you’re also using a case like the [recommended one from Sixfab](https://sixfab.com/product/raspberry-pi-ip54-outdoor-iot-project-enclosure/),
```

File: valve-monitor/README.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+your build can look like this when placed within the enclosure.
```
Comment (paigen11) [inline valve-monitor/README.md:117]: Consider including a note about _gently_ bending the antenna and leaving it on the outermost edge of the case to ensure a good cellular connection.

File: valve-monitor/README.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+![The full build within a Sixfab enclosure](images/full-build-in-case.jpg)
+
+With the hardware assembled, you next need to get the project’s firmware running on
```
Comment (paigen11) [inline valve-monitor/README.md:121]:
```suggestion
With the hardware assembled, the next step is to get the project’s firmware running on
```

File: valve-monitor/README.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+## Firmware
+
+The Valve Monitor project has custom firmware that runs on the Swan. As a next step,
+complete the [firmware’s setup guide](firmware/) to get your firmware up and running.
+
+## Web Application
+
+The Valve Monitor project’s web application allows you to view flow rates, open and
+close solenoid valves, and manage environment variables in a browser.
```
Comment (paigen11) [inline valve-monitor/README.md:132]:
```suggestion
close solenoid valves, set alarm thresholds, and manage environment variables in a browser.
```

File: valve-monitor/firmware/README.md [hunk 8]
```diff
@@ -1,41 +1,107 @@
...
-- Connect the F_D10 pin of the Notecarrier to your solenoid valve's control pin.
-- Connect the F_D6 pin of the Notecarrier to the signal pin of your flow rate
-meter.
-- Connect the Swan to your development machine with the micro USB cable.
-- Open Visual Studio Code and navigate to the PlatformIO menu using the icon
-on the leftmost side.
-- Select "Open" and open valve-monitor/firmware.
-- Edit main.cpp so to define a valid product UID:
+Once you have Visual Studio Code and the PlatformIO extension installed, you next need
+to open the firmware within PlatformIO.
```
Comment (paigen11) [inline valve-monitor/firmware/README.md:17]: Link to PlatformIO here?

File: valve-monitor/firmware/README.md [hunk 9]
```diff
@@ -1,41 +1,107 @@
...
+1. Select the `firmware` folder from within your local copy of this repository,
+and click the **Open "firmware"** button.
+![How to open the firmware folder in PlatformIO](../images/platformio-open-firmware.png)
+
+## Flashing
+
+With the firmware project open, you can now make any changes you’d like to the firmware
+source code, and then flash the firmware to your device.
+
+Complete the following steps to upload firmware in in PlatformIO.
```
Comment (paigen11) [inline valve-monitor/firmware/README.md:33]:
```suggestion
Complete the following steps to upload firmware in PlatformIO.
```

File: valve-monitor/firmware/README.md [hunk 10]
```diff
@@ -1,41 +1,107 @@
...
+
+Once the firmware is running on your device, you might additionally want to open a serial
+monitor to view the firmware’s logs. You can do so by clicking the plug button that appears
+at the bottom of your Visual Studio Code window.
+
+![Opening a serial monitor in PlatformIO](../images/platformio-serial-monitor.png)
+
+## Testing
+
+Once you have the firmware running, you’ll next want to the firmware’s `main.cpp` file,
```
Comment (paigen11) [inline valve-monitor/firmware/README.md:54]:
```suggestion
Once you have the firmware running, you’ll next want to open the firmware’s `main.cpp` file,
```

File: valve-monitor/firmware/README.md [hunk 11]
```diff
@@ -1,41 +1,107 @@
...
+```json
+"body": {
+    "state": "open" // or "close"
+}
+```
+
+> **NOTE**: If you’re unsure how to send Notes to a Notecard, check out this
+[tutorial on using the Notehub API](https://dev.blues.io/guides-and-tutorials/using-the-notehub-api/).
+
+The firmware additionally sends two types of outbound Notes.
```
Comment (paigen11) [inline valve-monitor/firmware/README.md:84]:
```suggestion
The firmware additionally sends two types of outbound Notes back to Notehub.
```

File: valve-monitor/firmware/README.md [hunk 12]
```diff
@@ -1,41 +1,107 @@
...
+(And if you don’t, refer back to the [hardware setup guide](../#hardware) for details
+on how to hook everything up).
+
+- Connect the `ATTN` and `F_D13` pins on the Notecarrier.
+- Connect the `F_D10` pin of the Notecarrier to your solenoid valve's control pin.
+- Connect the `F_D6` pin of the Notecarrier to the signal pin of your flow-rate
+meter.
+
+With everything connected and running, the firmware expects to receive `data.qi`
+Notes in the following format to open and close the solenoid valve.
```
Comment (paigen11) [inline valve-monitor/firmware/README.md:73]:
```suggestion
Notes from Notehub in the following format to open and close the solenoid valve.
```

File: valve-monitor/web-app/README.md [hunk 13]
```diff
@@ -0,0 +1,366 @@
+# Valve Monitor Web App
+
+The Valve Monitor project’s web application allows you to view flow rates, open and
+close solenoid valves, and manage environment variables in a browser.
```
Comment (paigen11) [inline valve-monitor/web-app/README.md:4]:
```suggestion
close solenoid valves, set flow rate alarm thresholds, and manage environment variables in a browser.
```

File: valve-monitor/web-app/README.md [hunk 14]
```diff
@@ -0,0 +1,366 @@
...
+To get started, make sure you have a copy of this project’s repository locally, open
+up the `valve-monitor/web-app` folder in your text editor or IDE, and then complete
+the following steps.
+
+- [Dependencies](#dependencies)
+  - [(Recommended) Visual Studio Code Dev Container](#recommended-visual-studio-code-dev-container)
+  - [(Not Recommended) Dependencies without VS Code](#not-recommended-dependencies-without-vs-code)
+- [Configuration (Environment Variables)](#configuration-environment-variables)
+  - [HUB_AUTH_TOKEN](#hub_auth_token)
+  - [HUB_PROJECT_UID](#hub_project_uid)
```
Comment (paigen11) [inline valve-monitor/web-app/README.md:17]: We have `HUB_FLEET_UID` as a config also, but it looks like it's never used in the project. I suggest removing from `Config.ts` and `.env.example` if that's the case.

File: valve-monitor/web-app/README.md [hunk 15]
```diff
@@ -0,0 +1,366 @@
...
+With the development server running, open <http://localhost:4000> in your web
+browser to see your application.
+
+Next.js automatically watches your project’s files, and updates your application
+as you make changes. To try it, open your app’s `src/pages/index.tsx` file, make
+a change, save the file, and notice how your browser automatically updates with
+the change. Changes to `.env` are **not** automatically reloaded, and require you
+to stop `yarn dev` with `ctrl+c`, and to then start `yarn dev` back up.
+
+The project’s `src/pages/api` directory are
```
Comment (paigen11) [inline valve-monitor/web-app/README.md:357]:
```suggestion
The project’s `src/pages/api` directory are for accessing outside
```

File: valve-monitor/web-app/README.md [hunk 16]
```diff
@@ -0,0 +1,366 @@
...
+[APIs](https://nextjs.org/docs/api-routes/introduction) as opposed to
+React-powered HTML pages.
```
Comment (paigen11) [inline valve-monitor/web-app/README.md:359]:
```suggestion
The project’s `src/pages/api` directory are
[APIs](https://nextjs.org/docs/api-routes/introduction) for fetching via the server as opposed to
React-powered client side HTML pages.
```

**Review by paigen11 — APPROVED** [review]
Minor suggestions for language and clarification, but overall great job. Really nice and straightforward

File: valve-monitor/firmware/README.md (hunk 8 above)
Comment (tjvantoll) [reply valve-monitor/firmware/README.md:17]: I linked to it at the top of this README which I think is probably sufficient.

File: valve-monitor/firmware/README.md (hunk 9 above)
Comment (tjvantoll) [reply valve-monitor/firmware/README.md:33]: 🙈

**Comment by tjvantoll** [comment]
@paigen11 Thanks! And good suggestions as always.

**Pull request merged**